### PR TITLE
clean up app.tsx

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

File: app/ts/components/PreactUtils.tsx
Comment (MicahZoltu) [inline app/ts/components/PreactUtils.tsx:1]: This should probably be moved to `ts/libraries/preact-utilities.ts`.  It isn't a component, and doesn't have any TSX in it.

**Review by MicahZoltu — APPROVED** [review]

File: app/css/index.css [hunk 1]
```diff
@@ -100,6 +100,8 @@ body {
 	border-radius: 25px;
 	padding: 20px;
 	box-shadow: 0 0 20px rgb(180, 60, 66);
+	margin: 0px;
+	margin-bottom: 20px;
```
Comment (jubalm) [inline app/css/index.css]:
```suggestion
	margin: 0 0 20px 0;
```

File: app/css/index.css [hunk 2]
```diff
@@ -111,6 +113,14 @@ body {
 	border-radius: 25px;
 	padding: 20px;
 	box-shadow: 0 0 20px #3cb371;
+	margin: 0px;
+	margin-bottom: 20px;
```
Comment (jubalm) [inline app/css/index.css]:
```suggestion
	margin: 0 0 20px 0;
```

File: app/ts/components/PreactUtils.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+				if (this.inner === undefined) this.inner = new Signal(newValue)
+				else this.inner.value = newValue
+				this.value = this.inner
+			})
+		}
+	}
+}
+
+export function useOptionalSignal<T>(value: Signal<T> | T | undefined, startUndefined?: boolean) {
+	return useMemo(() => new OptionalSignal<T>(value, startUndefined), []);
```
Comment (jubalm) [inline app/ts/components/PreactUtils.tsx:50]: i don't think `useMemo` does any memoizing here having a blank array dependency, (which means it's not going to update it's value)

```suggestion
	return new OptionalSignal<T>(value, startUndefined);
```

File: app/ts/components/requirements.tsx [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+				<label class = 'form-control'>
+					<p class = 'paragraph requirement'> 2) </p>
+					<input type = 'checkbox' name = 'switch' class = 'check' checked = { requirementsMet } disabled = { true }/>
+					<p class = 'paragraph checkbox-text requirement'> { childDomainInfo.value.label } exists </p>
+				</label>
+				<SwitchAddress requirementsMet = { requirementsMet } account = { account } signingAddress = { correctSigningAddress }/>
+			</div>
+		</div>
+		<div class = 'grid-item' style = 'justify-self: end; justify-content: center;'>
+			<button class = 'button is-primary' { ...buttonDisabled ? { disabled: true } : {} } onClick = { button }>
```
Comment (jubalm) [inline app/ts/components/requirements.tsx:74]:
```suggestion
			<button class = 'button is-primary' disabled = { buttonDisabled } onClick = { button }>
```

File: app/ts/components/requirements.tsx [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+					<p class = 'paragraph requirement'> 3) </p>
+					<input type = 'checkbox' name = 'switch' class = 'check' checked = { requirementsMet } disabled = { true }/>
+					<p class = 'paragraph checkbox-text requirement'> { childDomainInfo.value.label } is wrapped </p>
+				</label>
+				<p class = 'paragraph dim'> { `Requires approve all from the owner (asked if needed)` } </p>
+				<SwitchAddress requirementsMet = { requirementsMet } account = { account } signingAddress = { correctSigningAddress }/>
+			</div>
+		</div>
+		<div class = 'grid-item' style = 'justify-self: end; justify-content: center;'>
+			<button class = 'button is-primary' { ...buttonDisabled ? { disabled: true } : {} } onClick = { button }>
```
Comment (jubalm) [inline app/ts/components/requirements.tsx:99]:
```suggestion
			<button class = 'button is-primary' disabled = { buttonDisabled } onClick = { button }>
```

File: app/ts/components/requirements.tsx [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+						disabled = { checkBoxes.value?.childContentHashIsSet === true }
+						value = { contentHashInput.value } 
+						onInput = { e => handleContentHashInput(e.currentTarget.value) }
+					/>
+				<SwitchAddress requirementsMet = { requirementsMet } account = { account } signingAddress = { correctSigningAddress }/>
+				</> }
+			</div>
+		</div>
+		<div class = 'grid-item' style = 'justify-self: end; justify-content: center;'>
+			<button class = 'button is-primary' { ...buttonDisabled ? { disabled: true } : {} } onClick = { button }>
```
Comment (jubalm) [inline app/ts/components/requirements.tsx:147]:
```suggestion
			<button class = 'button is-primary' disabled = { buttonDisabled } onClick = { button }>
```

File: app/ts/components/requirements.tsx [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+					<p class = 'paragraph requirement'> 5) </p>
+					<input type = 'checkbox' name = 'switch' class = 'check' checked = { requirementsMet } disabled = { true }/>
+					<p class = 'paragraph checkbox-text requirement'> { parentDomainInfo.value.label } fuses are burnt </p>
+				</label>
+				<p class = 'paragraph dim'> { `The fuse "${ parentFuseToBurn }" is burnt` } </p>
+				<SwitchAddress requirementsMet = { requirementsMet } account = { account } signingAddress = { correctSigningAddress }/>
+			</div>
+		</div>
+		<div class = 'grid-item' style = 'justify-self: end; justify-content: center;'>
+			<button class = 'button is-primary' { ...buttonDisabled ? { disabled: true } : {} } onClick = { button }>
```
Comment (jubalm) [inline app/ts/components/requirements.tsx:173]:
```suggestion
			<button class = 'button is-primary' disabled = { buttonDisabled } onClick = { button }>
```

File: app/ts/components/requirements.tsx [hunk 8]
```diff
@@ -0,0 +1,230 @@
...
+					<p class = 'paragraph requirement'> 6) </p>
+					<input type = 'checkbox' name = 'switch' class = 'check' checked = { requirementsMet === true } disabled = { true }/>
+					<p class = 'paragraph checkbox-text requirement'> { childDomainInfo.value.label } fuses are burnt </p>
+				</label>
+				<p class = 'paragraph dim'> { `The fuses ${ childFusesToBurn.map((n) => `"${ n }"`).join(', ') } are burnt` } </p>
+				<SwitchAddress requirementsMet = { requirementsMet } account = { account } signingAddress = { correctSigningAddress }/>
+			</div>
+		</div>
+		<div class = 'grid-item' style = 'justify-self: end; justify-content: center;'>
+			<button class = 'button is-primary' { ...buttonDisabled ? { disabled: true } : {} } onClick = { button }>
```
Comment (jubalm) [inline app/ts/components/requirements.tsx:199]:
```suggestion
			<button class = 'button is-primary' disabled = { buttonDisabled } onClick = { button }>
```

File: app/ts/components/requirements.tsx [hunk 9]
```diff
@@ -0,0 +1,230 @@
...
+					<p class = 'paragraph requirement'> 7) </p>
+					<input type = 'checkbox' name = 'switch' class = 'check' checked = { requirementsMet } disabled = { true }/>
+					<p class = 'paragraph checkbox-text requirement'> { childDomainInfo.value.label } ownership is burnt </p>
+				</label>
+				<p class = 'paragraph dim'> { `The ownership of subdomain is moved to an address controlled by nobody` } </p>
+				<SwitchAddress requirementsMet = { requirementsMet } account = { account } signingAddress = { correctSigningAddress }/>
+			</div>
+		</div>
+		<div class = 'grid-item' style = 'justify-self: end; justify-content: center;'>
+			<button class = 'button is-primary' { ...buttonDisabled ? { disabled: true } : {} } onClick = { button }>
```
Comment (jubalm) [inline app/ts/components/requirements.tsx:225]:
```suggestion
			<button class = 'button is-primary' disabled = { buttonDisabled } onClick = { button }>
```

**Review by jubalm — APPROVED** [review]

File: app/ts/components/PreactUtils.tsx (hunk 3 above)
Comment (MicahZoltu) [reply app/ts/components/PreactUtils.tsx:50]: IIUC, the `[]` parameter is saying that the memoized result should *always* be returned, and this function should never re-run.  If you pass something to the second parameter, it tells the `useMemo` function to re-run if the contents of the array changes.

**Pull request merged**